### PR TITLE
8334653: ISO 4217 Amendment 177 Update

### PR DESCRIPTION
**Pull request by Karm**
backport b05f58e JDK11 JDK-8334653 ISO 4217 Amendment 177 Update
depends on #561

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] [JDK-8334653](https://bugs.openjdk.org/browse/JDK-8334653) needs maintainer approval

### Issue
 * [JDK-8334653](https://bugs.openjdk.org/browse/JDK-8334653): ISO 4217 Amendment 177 Update (**Bug** - P3 - Approved)

### Reviewers
 * [Andrew John Hughes](https://openjdk.org/census#andrew) (@gnu-andrew - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u-dev.git pull/562/head:pull/562` \
`$ git checkout pull/562`

Update a local copy of the PR: \
`$ git checkout pull/562` \
`$ git pull https://git.openjdk.org/jdk8u-dev.git pull/562/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 562`

View PR using the GUI difftool: \
`$ git pr show -t 562`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u-dev/pull/562.diff">https://git.openjdk.org/jdk8u-dev/pull/562.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk8u-dev/pull/562#issuecomment-2288835086)

### CODE REVIEW

**Comment by jerboaa** [comment]
@Karm Please merge master into this PR, resolve conflicts etc. Thanks!

**Comment by Karm** [comment]
Cleaned, reconfigured, rebuilt, tested:
```
$ TIME=`date +%s`;mkdir -p test.${TIME}/jdk/JTwork test.${TIME}/jdk/JTreport; jtreg -a -ignore:quiet -w:test.${TIME}/jdk/JTwork -r:test.${TIME}/jdk/JTreport -jdk:/home/karm/workspaceRH/jdk8u/build/linux-x86_64-normal-server-release/images/j2sdk-image/ /home/karm/workspaceRH/jdk8u/jdk/test/java/util/Currency/
Test results: passed: 5
Report written to /home/karm/Tools/jtreg/build/images/jtreg/bin/test.1724236868/jdk/JTreport/html/report.html
Results written to /home/karm/Tools/jtreg/build/images/jtreg/bin/test.1724236868/jdk/JTwork
```

**Comment by Karm** [comment]
/approval request Backporting regular ISO 4217 update from JDK 11. The backport adds only the new currency update, keeping the `FILEVERSION=2`, i.e. not dragging along other updates. Tests in jdk/test/java/util/Currency/ pass.

**Comment by jerboaa** [comment]
> /approval request Backporting regular ISO 4217 update from JDK 11. The backport adds only the new currency update, keeping the `FILEVERSION=2`, i.e. not dragging along other updates. Tests in jdk/test/java/util/Currency/ pass.

@Karm In general, please only apply for approval once you have a successful review. I.e. review first, then approval. Thanks! It's OK this time.

**Comment by Karm** [comment]
> > /approval request Backporting regular ISO 4217 update from JDK 11. The backport adds only the new currency update, keeping the `FILEVERSION=2`, i.e. not dragging along other updates. Tests in jdk/test/java/util/Currency/ pass.
> 
> @Karm In general, please only apply for approval once you have a successful review. I.e. review first, then approval. Thanks! It's OK this time.

Ack. Thx.

File: jdk/src/share/classes/java/util/CurrencyData.properties [hunk 1]
```diff
@@ -55,9 +55,8 @@ all=ADP020-AED784-AFA004-AFN971-ALL008-AMD051-ANG532-AOA973-ARS032-ATS040-AUD036
     TPE626-TRL792-TRY949-TTD780-TWD901-TZS834-UAH980-UGX800-USD840-USN997-USS998-UYI940-\
     UYU858-UZS860-VEB862-VED926-VEF937-VES928-VND704-VUV548-WST882-XAF950-XAG961-XAU959-XBA955-\
     XBB956-XBC957-XBD958-XCD951-XCG532-XDR960-XFO000-XFU000-XOF952-XPD964-XPF953-\
-    XPT962-XSU994-XTS963-XUA965-XXX999-YER886-YUM891-ZAR710-ZMK894-ZMW967-ZWD716-ZWL932-\
-    ZWN942-ZWR935
-
```
Comment (gnu-andrew) [inline jdk/src/share/classes/java/util/CurrencyData.properties:60]: This blank line should not be removed here.

**Review by gnu-andrew — COMMENTED** [review]
An additional blank line is removed in `CurrencyData.properties` and the bug ID line change is missing for `ValidateISO4217.java`. Otherwise, the backport matches the changes in the 11u version.

Have you verified that the updated Currency tests pass against the patched build?

**Comment by Karm** [comment]
> An additional blank line is removed in `CurrencyData.properties` and the bug ID line change is missing for `ValidateISO4217.java`. Otherwise, the backport matches the changes in the 11u version.
> 
> Have you verified that the updated Currency tests pass against the patched build?

https://github.com/openjdk/jdk8u-dev/pull/562#issuecomment-2301743430

i.e. yes I have, the Currency package.

File: jdk/src/share/classes/java/util/CurrencyData.properties (hunk 1 above)
Comment (Karm) [reply jdk/src/share/classes/java/util/CurrencyData.properties:60]: Will amend presently.

**Comment by Karm** [comment]
> `CurrencyTest.java` is missing this hunk:
> 
> ```
> @@ -60,7 +60,7 @@ import static org.junit.jupiter.api.Assertions.assertThrows;
> 
>  public class CurrencyTest {
> 
> -    // 'tablea1.txt' should be up-to-date before testing
> +    // 'ISO4217-list-one.txt' should be up-to-date before testing
>      @Test
>      public void dataVersionTest() {
>          CheckDataVersion.check();
> ```
> 
> See: [openjdk/jdk11u-dev@b05f58e#diff-9b35965750742d3e80baa3c18194cd60156bf561990636e49fa0f6eadf4854dfR63](https://github.com/openjdk/jdk11u-dev/commit/b05f58e9ec57ce7f140ad47df3b5bdf4cc42f5f9#diff-9b35965750742d3e80baa3c18194cd60156bf561990636e49fa0f6eadf4854dfR63)

@jerboaa Why do you think that? I did not port the tests to JUnit, I left that in its original state. The data file is loaded in CheckDataVersion. No point in bringing the comment that belongs to a JUnit that does not exist here IMHO.

**Comment by Karm** [comment]
Or perhaps I misunderstood something.

**Comment by gnu-andrew** [comment]
> Or perhaps I misunderstood something.

No, this is the same conclusion I came to. 8u does not have [JDK-8310923] (https://bugs.openjdk.org/browse/JDK-8310923) which completely refactors this test and is part of a bigger task to move to a newer version of JTreg across the testsuite. We don't intend to backport that to 8u.

**Review by gnu-andrew — APPROVED** [review]
Thanks for correcting those two issues. Looks good to me now.

**Pull request closed without merging**